src/web/client/utilities/graphClientUtils.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import path from "path";
+import WebExtensionContext from "../WebExtensionContext";
+import { getCommonHeaders } from "../common/authenticationProvider";
+import * as Constants from "../common/constants";
+import { telemetryEventNames } from "../telemetry/constants";
+
+export async function requestGraphClient(graphToken: string, service: string, userId: string) {
+    let requestSentAtTime = new Date().getTime();
```

Comment:
one recommendation is to create a class for all these graph functions - refer telemetry class example. You can initialize that class in webExtensionContext and store the graph token in that class as private. Only functions inside that class will then access the token to make graph calls - and the whole extension wont have acccess to this token.